### PR TITLE
OCNE cluster creation troubleshooting

### PR DESCRIPTION
**Pull request by rhomast**

### CODE REVIEW

**Comment by rhomast** [comment]
VCN requirements will be added in a later commit

File: content/en/docs/troubleshooting/troubleshooting-clusterapi.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+```
+kubectl logs -n verrazzano-capi capi-ocne-bootstrap-controller-manager-<nnnn>
+```
+{{< /clipboard >}}
+</div>
+
+**Review clusterAPI Kubernetes custom resources**
+
+On the admin cluster, use `kubectl` to check the status of the clusterAPI custom resource:
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-clusterapi.md:66]: Probably it should be a period at the end of this first sentence.
-> clusterAPI custom resource.

File: content/en/docs/troubleshooting/troubleshooting-clusterapi.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+You can check the status of the required OCI resources in the OCI console or by reviewing the log files for the clusterAPI controller.
+
+Causes may include:
+
+* OCI resource limits were reached: Cluster creation will fail if it needs to create more resources than allowed by your OCI tenancy. Check your OCI service limits to see if you exceeded its limits. If you need to request a service limit increase, see [Requesting a Service Limit Increase](https://docs.oracle.com/en-us/iaas/Content/General/Concepts/servicelimits.htm#Requesti).
+
+* Connectivity issues: Make sure your virtual cloud network is configured properly.
+
+* The network load balancer is in a critical state: The network load balancer may temporarily enter a critical state during initial cluster creation until the Kubernetes API server is up. If it remains in a critical state, then one of the following issues may have occurred:
+    * Traffic between the network load balancer and the OCNE control plane node is blocked over port 6443
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-clusterapi.md:110]: Need a period at the end of each bulleted item in this list.

File: content/en/docs/troubleshooting/troubleshooting-clusterapi.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+    * The OCNE API server did not start, possibly because OCNE dependencies failed to install
+
+* Node creation failed:
+    * The OCI credentials are invalid: Check for errors under Kubernetes events in the namespace where cluster objects are present or in the OCI provider log files.
+
+    * The Image ID used to deploy templates is invalid.
+
+    * Worker nodes only: Worker nodes began provisioning after the control node plane entered `Running` state.
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-clusterapi.md:119]: -> entered the `Running` state.

File: content/en/docs/troubleshooting/troubleshooting-clusterapi.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        * If OCNE dependencies failed to install on the control plane node, worker nodes remain in Pending state and do not get created. Check the `cloud-init` log files on control plane nodes to determine the cause.
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-clusterapi.md:120]: -> worker nodes remain in the `Pending` state
-> log files on the control plane nodes

File: content/en/docs/troubleshooting/troubleshooting-clusterapi.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+You should also confirm that pods are running on the workload cluster. From the console, download the kubeconfig file for the workload cluster and run the following command:
```
Comment (rosemarymarano) [inline content/en/docs/troubleshooting/troubleshooting-clusterapi.md:122]: -> Also, you should confirm (so as to not split the verb)

**Review by rosemarymarano — APPROVED** [review]

File: content/en/docs/troubleshooting/troubleshooting-clusterapi.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+* Connectivity issues: Make sure your virtual cloud network is configured properly.
+
+* The network load balancer is in a critical state: The network load balancer may temporarily enter a critical state during initial cluster creation until the Kubernetes API server is up. If it remains in a critical state, then one of the following issues may have occurred:
+    * Traffic between the network load balancer and the OCNE control plane node is blocked over port 6443.
+
+    * The OCNE API server did not start, possibly because OCNE dependencies failed to install.
+
+* Node creation failed:
+    * The OCI credentials are invalid: Check for errors under Kubernetes events in the namespace where cluster objects are present or in the OCI provider log files.
```
Comment (bobdonat) [inline content/en/docs/troubleshooting/troubleshooting-clusterapi.md:115]: This is the command to look at the OCI provider log file.  There is only one.
kubectl logs -n verrazzano-capi capoci-controller-manager-<nnnn>

File: content/en/docs/troubleshooting/troubleshooting-clusterapi.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+If one of these resources was not created, or was created but is now experiencing issues, it can affect the rest of the cluster.
+
+You can check the status of the required OCI resources in the OCI console or by reviewing the log files for the clusterAPI controller.
+
+Causes may include:
+
+* OCI resource limits were reached: Cluster creation will fail if it needs to create more resources than allowed by your OCI tenancy. Check your OCI service limits to see if you exceeded its limits. If you need to request a service limit increase, see [Requesting a Service Limit Increase](https://docs.oracle.com/en-us/iaas/Content/General/Concepts/servicelimits.htm#Requesti).
+
+* Connectivity issues: Make sure your virtual cloud network is configured properly.
```
Comment (bobdonat) [inline content/en/docs/troubleshooting/troubleshooting-clusterapi.md:107]: should there be a reference to doc on configuring a VCN properly

File: content/en/docs/troubleshooting/troubleshooting-clusterapi.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+
+If you encounter a problem with the OCNE clusters created using the clusterAPI component, here are some techniques you can use to diagnose and solve the issue.
+
+### Assess the condition of the clusterAPI component
+
+You can gather information about the clusterAPI component by reviewing its log files and by checking the status of clusterAPI custom resources on the admin cluster. In the console, the admin cluster is the *local* cluster.
+
+**Review clusterAPI log files**
+
+The clusterAPI component creates four clusterAPI pods within the `verrazzano-capi` namespace and each pod generates a separate log file. Review the log files to determine the location of the issue.
```
Comment (bobdonat) [inline content/en/docs/troubleshooting/troubleshooting-clusterapi.md:16]: "location" does not seem the right word here

File: content/en/docs/troubleshooting/troubleshooting-clusterapi.md [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+### Verify the status of OCI resources
+
+When creating clusters, Verrazzano creates the following resources in the associated OCI compartment:
+
+* [A network load balancer](https://docs.oracle.com/en-us/iaas/Content/NetworkLoadBalancer/NetworkLoadBalancers/list-network-load-balancer.htm)
+* [Compute instances for each control plane node and worker node](https://docs.oracle.com/en-us/iaas/Content/Compute/home.htm)
+
+If one of these resources was not created, or was created but is now experiencing issues, it can affect the rest of the cluster.
+
+You can check the status of the required OCI resources in the OCI console or by reviewing the log files for the clusterAPI controller.
```
Comment (bobdonat) [inline content/en/docs/troubleshooting/troubleshooting-clusterapi.md:101]: should this reference the review log files section abover

**Comment by rhomast** [comment]
@theabmitra I've added the documentation for the VCN requirements.

Do you think it would be helpful to incorporate the examples on this page https://verrazzano.io/latest/docs/setup/install/prepare/platforms/olcne/#examples into this section?

**Pull request merged**